hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/publish/TransactionPublisher.java

```diff
@@ -56,12 +60,15 @@
 
     private final MonitorProperties monitorProperties;
     private final PublishProperties publishProperties;
-    private final List<AccountId> nodeAccountIds = new CopyOnWriteArrayList<>();
+    private final AtomicReference<List<AccountId>> nodeAccountIds = new AtomicReference<>(List.of());
     private final Flux<Client> clients = Flux.defer(this::getClients).cache();
     private final SecureRandom secureRandom = new SecureRandom();
+    private Optional<Disposable> nodeValidator = Optional.empty();
```

Comment:
```suggestion
    private volatile Optional<Disposable> nodeValidator = Optional.empty();
```
Or we can swap it to use AtomicReference instead of Optional.

Reply:
Added

Reply:
Swapped to `AtomicReference` to make Sonarcloud happy, no bugs now.